### ISSUE TITLE
Online status issue

### ISSUE THREAD

**Issue by MukeshSherawat61** [RocketChat/Rocket.Chat#31768]
Online status not showing team members when they use rocket chat desktop app.

**Comment by Gummikavalier** [comment]
Has been fixed but not released according to https://github.com/RocketChat/Rocket.Chat/issues/31705

**Comment by lovekyleman** [comment]
I updated RC instances from 6.6.0 to 6.6.1. The bug isn't fixed

**Comment by lovekyleman** [comment]
> I updated RC instances from 6.6.0 to 6.6.1. The bug isn't fixed 
> 

Users online that shows offline and user offline that shows online

**Comment by Gummikavalier** [comment]
Works on our test RC instance.

Did you remember to do shift-reload on both web clients, and then try switching statuses again?

**Comment by lovekyleman** [comment]
It doesn't work on Rocket.Chat iOS app.

**Comment by sampaiodiego** [comment]
can you guys that are still having the issue describe more about your deployment pls?

- how many rocketchat instances?
- monolith or micro services?
- server specs
- mongodb deployment (3 node replica set? atlas?)
- using change streams? oplog? any special env var?
- what was your previous version where this was working?
- how are you testing it out?
- please fill out the [bug report template](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/ISSUE_TEMPLATE/bug_report.md)

the more details the better..

the issue on 6.6.0 was only related to the Web UI not showing the updated status (the status circle was not getting updated), but on page the refresh the status were correct.. if the issue you guys are seeing is different than this, we need to understand what is going on in order to fix it too.

**Comment by lovekyleman** [comment]
Description:iOS app shows all users always online 

Steps to reproduce:

Expected behavior:when user offline it should display offline 

Actual behavior:users online forever 

Server Setup Information:

Version of Rocket.Chat Server:6.6.1
Operating System:Debian 11
Deployment Method:manually 
Number of Running Instances:3
DB Replicaset Oplog:yes 
NodeJS Version:14.21.3
MongoDB Version:6.0.13
Client Setup Information

Desktop App or Browser Version:iOS app 4.46.1.57000 ( the latest)
Operating System:
Additional context

Relevant logs:no logs

**Comment by Gummikavalier** [comment]
I too can confirm presence issue in the iOS client 4.46.1.57000.

Restarting the app or clearing the cache app resets the correct presence information, until it is changed by the user, at which point the user's status goes offline in the app.

Regardless of the selected status by this user, for instance choosing `Away` changes this user's status into `Offline` status for the others.

**Comment by sampaiodiego** [comment]
@diegolmello there seems to be an issue with how the mobile app is handling the presence updates.

**Comment by diegolmello** [comment]
> Restarting the app or clearing the cache app resets the correct presence information, until it is changed by the user, at which point the user's status goes offline in the app.

I was able to reproduce that. Going to take a look in a few.

**Comment by chotaire** [comment]
I'm still facing the same issue with people showing as online (green) even when their devices are turned off since version 6.6.0. I thought it was fixed in 6.6.1, but the problem just re-appeared over the weekend.

**Comment by TimmiORG** [comment]
This should be the fix for the issue: https://github.com/RocketChat/Rocket.Chat/pull/31823
Hopefully there will be a 6.6.2 soon.

**Comment by sampaiodiego** [comment]
https://github.com/RocketChat/Rocket.Chat/pull/31823 is actually already shipped on 6.6.1 ..

but there is a follow up fix https://github.com/RocketChat/Rocket.Chat/pull/31833 that will be shipped on 6.6.2 soon..

**Comment by sampaiodiego** [comment]
I'm sorry, https://github.com/RocketChat/Rocket.Chat/pull/31823 is not actually on 6.6.1 .. I'll include it in 6.6.2 along with https://github.com/RocketChat/Rocket.Chat/pull/31833 today

**Comment by Gummikavalier** [comment]
@diegolmello @sampaiodiego Other issues are fixed but there still is one more issue I was able to reproduce with RC 6.6.2:

On one of our RC setups I was not able to get my presence go to offline by closing down the browser that last the session to RC server.
I was checking with another account that it remained online/green.

Finally I did `Logout from other logged in locations` under my `/account/profile`. This fixed the status.

The status reacted ok to browser closing and starting until I logged back into the RC mobile client, and chose `busy`.

At this point the issue with the offline status begun again.

**Comment by Gummikavalier** [comment]
@diegolmello @sampaiodiego Other issues are fixed but there still is one more issue I was able to reproduce with RC 6.6.2:

On one of our RC setups I was not able to get my presence go to offline by closing down the browser that last the session to RC server.
I was checking with another account that it remained online/green.

Finally I did `Logout from other logged in locations` under my `/account/profile` and then I logged out from this last browser session. This fixed the status operation.

After this the status reacted ok to browser closing and starting, until I logged back into the RC mobile client, and chose `busy`.

At this point the issue with the offline status begun again.

**Comment by chotaire** [comment]
> some of those notifications are dependent on user's presence, so maybe.. need to understand exactly what is going on to have a proper guess.

Tested 6.6.2. Presence still broken in the way I can log myself out of the web session and out of the android app and I will still show online to everyone. Notifications still entirely broken for my account on both Web and Android. I even deleted cache and relogged on both architectures, absolutely no change at all. On Android, I can't even seem to preview any sounds I would set for a channel or DM, nothing is ever playing. This notification problem is soon becoming a 100% showstopper to me, I will be unable to travel with this issue and if I can't find any solution for this, I'll unfortunately have no other choice than to abandon the project.

**Comment by lovekyleman** [comment]
6.6.2 doesn't fix the issue

**Comment by xcpdq** [comment]
I confirm this issue still persists for us even with 6.6.2 version. During these hours it's impossible to have so many people online in our team.
![WzW6BJjOaf6k6iI5VUM5-rc](https://github.com/RocketChat/Rocket.Chat/assets/18256939/6d175877-a205-4e82-b957-e28fca5fab4f)

**Comment by pythys** [comment]
Yup, I also confirm the bug persists in 6.6.2

**Comment by danel1** [comment]
Confirm, this is still an issue. @sampaiodiego That's some core functionality, which is quite essential. Any update?

**Comment by pythys** [comment]
We've bumped two releases and the issue still persists for something which should be considered fundamental behavior. I've learned a few lessons here:

- do not upgrade to a new minor / major version until it gets stabilized.
- test deployments as production releases might sometimes exhibit issues.
- keep an eye on critical bugs or issues before upgrading.

Is downgrade possible folks? Did anyone attempt it or are there changes that persisted in the database?
